src/index.ts

```diff
@@ -5,21 +5,13 @@ export * as Errors from './errors';
 
 // Type exports
 export type {
-  CollectionDescription,
```

Comment:
Most of the removals here were in lieu of types we're now relying on from the generated core. I think we probably need a more thorough pass of what we're exporting though.